azerothcore/azerothcore-wotlk
# feat(Core/Command): Use timestring with all commands

## Changes Proposed:
- `.mute` now uses TimeString inputs
- `.npc set spawntime` now uses TimeString inputs
- `.server shutdown|restart|idleshutdown|idlerestart` now use TimeString inputs
- `.bf timer` now uses TimeString inputs
- `.deserter bg|instance add` now use TimeString inputs
- `.deserter bg|instance add` time is now optional. Defaults to 15m for bg and 30m for instance
- Fixes a bug where you could use negative numbers on server commands
- Still supporting the old system with only numbers as input (defaulting to seconds)

## Issues Addressed:

- Closes #11920

## Tests Performed:

- Executed the modified commands with and without TimeString


## How to Test the Changes:


1. Run `.mute 1m`
2. Run `.mute 120`
3. Run `.npc set spawntime 600` and check with `.npc info`
4. Run `.npc set spawntime 10m` and check with `.npc info`
5. Run `.server shutdown|restart|idleshutdown|idlerestart 1m` and observe the broadcast
6. Run `.server shutdown|restart|idleshutdown|idlerestart 120` and observe the broadcast
7. Run  `.bf timer 5m`
8. Run `.bf timer 50`
9. Run `.deserter bg|instance add 30`
10. Run `.deserter bg|instance add 5m`

## Known Issues and TODO List:


- [x] `.mute`
- [x] `.deserter`
    - [x] ~remove all - requires~ included in #11963
    - [x] shutdown
    - [x] restart
    - [x] idleshutdown
    - [x] idlerestart
- [x] `.npc set spawntime`
- [x] `.bf timer`


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of cod

## Changes Proposed:
- `.mute` now uses TimeString inputs
- `.npc set spawntime` now uses TimeString inputs
- `.server shutdown|restart|idleshutdown|idlerestart` now use TimeString inputs
- `.bf timer` now uses TimeString inputs
- `.deserter bg|instance add` now use TimeString inputs
- `.deserter bg|instance add` time is now optional. Defaults to 15m for bg and 30m for instance
- Fixes a bug where you could use negative numbers on server commands
- Still supporting the old system with only numbers as input (defaulting to seconds)

## Issues Addressed:

- Closes #11920

## Tests Performed:

- Executed the modified commands with and without TimeString


## How to Test the Changes:


1. Run `.mute 1m`
2. Run `.mute 120`
3. Run `.npc set spawntime 600` and check with `.npc info`
4. Run `.npc set spawntime 10m` and check with `.npc info`
5. Run `.server shutdown|restart|idleshutdown|idlerestart 1m` and observe the broadcast
6. Run `.server shutdown|restart|idleshutdown|idlerestart 120` and observe the broadcast
7. Run  `.bf timer 5m`
8. Run `.bf timer 50`
9. Run `.deserter bg|instance add 30`
10. Run `.deserter bg|instance add 5m`

## Known Issues and TODO List:


- [x] `.mute`
- [x] `.deserter`
    - [x] ~remove all - requires~ included in #11963
    - [x] shutdown
    - [x] restart
    - [x] idleshutdown
    - [x] idlerestart
- [x] `.npc set spawntime`
- [x] `.bf timer`


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
e that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
